src/components/studies/participants/BatchEditForm.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react'
+import { Dialog } from '@material-ui/core'
+import { EditParticipantForm } from './ParticipantForms'
+import { EditDialogTitle } from './ParticipantTableGrid'
+import ParticipantService from '../../../services/participants.service'
+
+type BatchEditFormProps = {
+  isEnrolledById: boolean
+  isBatchEditOpen: boolean
+  setIsBatchEditOpen: Function
+  selectedParticipants: string[]
+  token: string
+  studyId: string
+  toggleParticipantRefresh: Function
+  isAllSelected: boolean
+}
+
+const BatchEditForm: React.FunctionComponent<BatchEditFormProps> = ({
+  isEnrolledById,
+  isBatchEditOpen,
+  setIsBatchEditOpen,
+  selectedParticipants,
+  token,
+  studyId,
+  toggleParticipantRefresh,
+  isAllSelected,
+}) => {
+  const [isLoading, setIsLoading] = React.useState(false)
+
+  return (
+    <Dialog
+      open={isBatchEditOpen}
+      maxWidth="sm"
+      fullWidth
+      aria-labelledby="edit participant"
+    >
+      <EditDialogTitle
+        onCancel={() => {
+          setIsBatchEditOpen(false)
+        }}
+        shouldWithdraw={false}
+        batchEdit
+      />
+      <EditParticipantForm
+        isEnrolledById={isEnrolledById}
+        onCancel={() => {
+          setIsBatchEditOpen(false)
+        }}
+        onOK={async (note: string, cvd?: Date) => {
+          if (!note && !cvd) return
+          setIsLoading(true)
+          let participants = selectedParticipants
+          if (isAllSelected) {
+            const resultEnrolled = await ParticipantService.getAllParticipantsInEnrollmentType(
+              studyId,
+              token,
+              'enrolled',
+              false,
+            )
+            const enrolledNonTestParticipants = resultEnrolled.items
+            participants = enrolledNonTestParticipants.map(
+              el => el.participant.identifier,
+            )
+          }
+          for (const selectedParticipantId of participants) {
```

Comment:
you don't need to wait for one call to complete to make the next. So it would make more sense here to use map instead of the 'for' loop to transform the array of participantIds into array of promises and then do a single away for Promise.all
Same with updating clinic visit.
Also the action (if you are updating note or the date) would apply to all of the participants therefore it should be outside of the loop/map